lib/api/src/sys/externals/memory.rs

```diff
@@ -99,12 +101,73 @@ impl Memory {
         mem.copy()
     }
 
+    pub fn to_shared(&self, store: &impl AsStoreRef) -> Option<crate::SharedMemory> {
+        let mem = self.handle.get(store.as_store_ref().objects());
```

Comment:
We need to check that the memory is actually a shared memory.

Reply:
That already happened implicitly, because only shared memories can hand out a shared memory handle. 

But I added an additional sanity check regardless, can't hurt.